test/jdk/tools/jpackage/share/AddLauncherTest.java

```diff
@@ -200,6 +202,33 @@ public void testMainLauncherIsModular(boolean mainLauncherIsModular) {
         .applyTo(cmd);
 
         cmd.executeAndAssertHelloAppImageCreated();
+
+        // verify using each additional launchers cfg file that it has the
+        // main-jar/class or module/class
+
+        CfgFile cfg = cmd.readLauncherCfgFile("ModularAppLauncher");
+        String moduleValue = cfg.getValue("Application", "app.mainmodule");
+        String mainClass = null;
+        String classpath = null;
+        TKit.assertEquals(moduleValue, JavaAppDesc.parse(
+                modularAppDesc.toString()).setBundleFileName(null).toString(),
+                "app.mainmodule value in cfg file not as expected");
+
+        TKit.trace("moduleValue: " + moduleValue + " mainClass: " + mainClass
+                    + " classpath: " + classpath);
+
+        cfg = cmd.readLauncherCfgFile("NonModularAppLauncher");
+        moduleValue = null;
+        mainClass = cfg.getValue("Application", "app.mainclass");
+        classpath = cfg.getValue("Application", "app.classpath");
+        TKit.assertEquals(mainClass, nonModularAppDesc.className(),
```

Comment:
Two first parameters of TKit.assertEquals() call need to be swapped as the first parameter should be expected and the second actual value.